lib/aws-elasticache-serverless-stack.ts

```diff
@@ -1,9 +1,62 @@
 import * as cdk from 'aws-cdk-lib';
+import * as ec2 from 'aws-cdk-lib/aws-ec2';
 import { Construct } from 'constructs';
+import { aws_elasticache as ElastiCache } from "aws-cdk-lib";
+import { SecurityGroup } from "aws-cdk-lib/aws-ec2";
+import { aws_kms as kms } from "aws-cdk-lib";
+import { AwsElasticacheServerlessStackProps } from './AwsElasticacheServerlessStackProps';
+import { parseVpcSubnetType } from '../utils/vpc-type-parser';
 
 export class AwsElasticacheServerlessStack extends cdk.Stack {
-  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
+  constructor(scope: Construct, id: string, props: AwsElasticacheServerlessStackProps) {
     super(scope, id, props);
 
+    const vpc = ec2.Vpc.fromLookup(this, `${props.resourcePrefix}-VPC-Imported`, {
+      vpcId: props.vpcId,
+    });
+
+    const vpcSubnetType = parseVpcSubnetType(props.vpcSubnetType);
+    const elastiCacheSubnetIds = vpc.selectSubnets({
+      subnetType: vpcSubnetType,
+    }).subnetIds;
+
+    const elastiCacheSecurityGroup = new SecurityGroup(this, `${props.resourcePrefix}-ElastiCache-Security-Group`, {
+      vpc,
+      description: `${props.resourcePrefix}-ElastiCache-Security-Group`,
+      allowAllOutbound: true,
+    });
+
+    const kmsKey = new kms.Key(this, `${props.resourcePrefix}-KMS-Key`, {
+      description: `${props.resourcePrefix}-KMS-Key`,
+      enableKeyRotation: true,
+    });
+
+    const elastiCacheServerless = new ElastiCache.CfnServerlessCache(
+      this,
+      `${props.resourcePrefix}-ElastiCache-Serverless`,
+      {
+        engine: "redis",
+        serverlessCacheName: `${props.appName}-${props.deployEnvironment}`,
+        securityGroupIds: [elastiCacheSecurityGroup.securityGroupId],
+        subnetIds: elastiCacheSubnetIds,
+        kmsKeyId: kmsKey.keyId,
+        description: `${props.resourcePrefix}-ElastiCache-Serverless`,
+        majorEngineVersion: "7",
+      },
```

Comment:
**Suggestion:** Add tags to the ElastiCache Serverless instance for better resource management and cost tracking. This can be done by adding the `tags` property with the same tags used in the `ApplyTags` aspect.
> 🚀  **High Priority** 
 >[Category: **`Enhancement`**, Importance: **`8`**]
> **Reason:** Adding tags to the ElastiCache Serverless instance improves resource management and cost tracking, aligning with best practices for cloud resource governance. The suggestion is well-aligned with the existing tagging strategy in the PR and enhances the maintainability of the infrastructure.

```suggestion
    const elastiCacheServerless = new ElastiCache.CfnServerlessCache(
      this,
      `${props.resourcePrefix}-ElastiCache-Serverless`,
      {
        engine: "redis",
        serverlessCacheName: `${props.appName}-${props.deployEnvironment}`,
        securityGroupIds: [elastiCacheSecurityGroup.securityGroupId],
        subnetIds: elastiCacheSubnetIds,
        kmsKeyId: kmsKey.keyId,
        description: `${props.resourcePrefix}-ElastiCache-Serverless`,
        majorEngineVersion: "7",
        tags: [
          { key: 'environment', value: props.deployEnvironment },
          { key: 'project', value: props.appName },
          { key: 'owner', value: props.owner }
        ]
      },
    );
```
- [ ]  **I have reviewed this code suggestion and addressed the relevant ones.**